File: Binary Modulo - GFG/binary-modulo.java
//{ Driver Code Starts
//Initial Template for Java

import java.util.*;
import java.io.*;
import java.lang.*;

class GFG{
	public static void main(String [] args) throws IOException{
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int test = Integer.parseInt(br.readLine());
		while(test-- > 0) {
			String [] str = br.readLine().trim().split(" ");
			String s = str[0];
			int m = Integer.parseInt(str[1]);
			Solution obj = new Solution();
			System.out.println(obj.modulo(s,m));
		}
	}
}

// } Driver Code Ends


//User function Template for Java

class Solution{
	int modulo(String s, int m) {
	java.math.BigInteger b1 = new java.math.BigInteger(s,2);
	java.math.BigInteger b2 = new java.math.BigInteger(String.valueOf(m));
	return b1.mod(b2).intValue();
	}
}

File: Easy/Binary Modulo/binary-modulo.py
#User function Template for python3

class Solution():
    def modulo(self, s, m):
         k = int(s,2);
         r = k % m
         return r
        



#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == "__main__":
    for _ in range(int(input())):
        s,m = input().split()
        m = int(m)
        print(Solution().modulo(s, m))

# } Driver Code Ends